src/Symfony/Component/Validator/Constraints/Finite.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * @Annotation
+ *
+ * @Target({"PROPERTY", "METHOD", "ANNOTATION"})
+ *
+ * @author Guillaume Aveline <guillaume@codr.fr>
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+class Finite extends Constraint
+{
+    public const NOT_FINITE_ERROR = '5f809eb0-78b9-492d-ad37-5a5188390415';
+
+    protected const ERROR_NAMES = [
+        self::NOT_FINITE_ERROR => 'NOT_FINITE_ERROR',
+    ];
+
+    public $message = 'This value should be finite.';
+
+    public function __construct(string $message = null, array $groups = null, mixed $payload = null, array $options = [])
```

Comment:
to be compatible with the XML and Yaml loaders, the `options` argument must be the first one.

Reply:
Made the changes, and also found where ValidatorTests were.
Please tell me if I miss anything.